kotest-property/kotest-property-core/src/jvmTest/kotlin/io/kotest/property/core/DelegateTest.kt

```diff
@@ -0,0 +1,97 @@
+package io.kotest.property.core
+
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.string.shouldEndWith
+import io.kotest.matchers.string.shouldHaveLength
+import io.kotest.matchers.string.shouldStartWith
+import io.kotest.property.Arb
+import io.kotest.property.ShrinkingMode
+import io.kotest.property.arbitrary.int
+import io.kotest.property.arbitrary.string
+import kotlin.time.Duration.Companion.milliseconds
+
+class DelegateTest : FunSpec() {
+   init {
+      proptest("delegate syntax 1") {
+
+         iterations = 42 // run this property test 42 times
+         failOnSeed = true // will fail if a seed is set
+         edgecaseFactor = 0.1 // 10% of generated values will be edgecases
+         shouldPrintGeneratedValues = true // output each generated value
+
+         val a by gen { Arb.int() }
```

Comment:
I think it would be a bit cleaner if the syntax was just 

```kotlin
val a by Arb.int()
```

But maybe the `gen {}` is important?

Reply:
It means we can avoid the Arb.int() being created each time, if that's important. It also means I don't have to make any changes to the existing Arb interface.

Reply:
Ah yes, avoiding changes to the Arb interface makes sense.

To avoid creating the Arb every time, I think changing GenDelegate to implement  PropertyDelegateProvider would help, and make the code a bit simpler. But I always get confused by the delegate stuff so maybe I'm wrong.

Reply:
Would directly using `Arb.int()` as delegate really create it each time? I'd expect it to be only created once, just the getValue being called multiple times - or maybe I'm misunderstanding something?

Also, regarding the interface: you don't need to change the interface, but you can implement `getValue(...)` as an extension method within the test scope; see comment at GenDelegate.

In addition, it would be nice to have the delegate functionality on a scope level. I encountered a few cases with a lot of tests basically requiring the same kind of generated data over and over again. I helped myself by building a generator for a data class, but that felt rather ugly. Having a whole set of top level delegates that get lazily initialized as soon as being accessed would be perfect for those cases.

Reply:
I've been trying to make this work:

            val a by Arb.int()
            val b by Exhaustive.ints(0..10)

But I'm hitting a dead end. In order to reset the values between runs we need access to the delegate itself (not just the values produced). We can do that by adding a provideDelegate extension method to Gen's. But that method doesn't provide access to the implicit receiver, so we have no place to store the delegates. I tried with a global container, which works unless you run tests in parallel because it doesn't know which delegates are associated with which test.